src/components/RecipeList.tsx

```diff
@@ -1,21 +1,63 @@
-import React, { useEffect, useState } from "react";
+import React, { useEffect, useMemo, useState } from "react";
 import ListGroup from "react-bootstrap/ListGroup";
 import Spinner from "react-bootstrap/Spinner";
 import Collapse from "react-bootstrap/Collapse";
 import Alert from "react-bootstrap/Alert";
 import FullRecipe from "./FullRecipe/FullRecipe";
 import { useGetAllRecipesMutation } from "../services/RecipeService";
+import PaginationComponent from "./UI/PaginationComponent";
+import AmoutnRecipesFilter from "./AmountRecipesFilter";
+import { useDispatch, useSelector } from "react-redux";
+import { RootState } from "../redux/store";
+import { removeRecipes, setRecipes } from "../redux/recipeListSlice";
+import styled from "styled-components";
+
+const Btn = styled.button`
+  margin: 0.5rem;
+  background-color: #0d6efd;
+  border: 1px solid rgba(27, 31, 35, 0.2);
+  color: #fff;
+  width: 30%;
+  font-weight: 600;
+  padding: 6px 12px;
+  border-radius: 5px;
+  &:hover {
+    background-color: #0d65eb;
+  }
+`;
 
 const RecipeList: React.FC = () => {
-  const [getAllRecipes, { data: recipes, isLoading, isError }] =
-    useGetAllRecipesMutation();
+  const [getAllRecipes, { isLoading, isError }] = useGetAllRecipesMutation();
+  const [page, setPage] = useState(1);
+  const dispatch = useDispatch();
+
+  const recipeList = useSelector((state: RootState) => {
+    return state.recipes;
+  });
+
+  const fetchREcipes = () => {
+    getAllRecipes()
+      .unwrap()
+      .then((resp) => {
+        console.log(resp);
+        dispatch(setRecipes(resp));
+      });
+  };
 
   useEffect(() => {
-    getAllRecipes().unwrap();
+    if (recipeList.recipes.length < 1) {
+      fetchREcipes();
+    }
   }, []);
```

Comment:
from my linter:
React Hook useEffect has missing dependencies: 'fetchREcipes' and 'recipeList.recipes.length'. Either include them or remove the dependency array.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)

in this case it won't affect but let's imagine you have 10k recipes; you fetch them every time when user switch the page [from pagination] and useEffect has `page` as a useEffect param. you can do this or fetch all of them once; but if once, why not to do this in the parent element. if every time - in child component [with checking of we have recipes from that page in the cache already]

Reply:
and about linter - it's always good to read it; sometimes it's really annoying but in general it helps a lot with the common mistakes ;)